wandb/filesync/step_prepare.py

```diff
@@ -18,29 +17,10 @@
 if TYPE_CHECKING:
     from wandb.sdk.internal import internal_api
 
-    if sys.version_info >= (3, 8):
-        from typing import Protocol
-    else:
-        from typing_extensions import Protocol
-
-    class DoPrepareFn(Protocol):
-        def __call__(self) -> "internal_api.CreateArtifactFileSpecInput":
-            pass
-
-    class OnPrepareFn(Protocol):
-        def __call__(
-            self,
-            upload_url: Optional[str],  # GraphQL type File.uploadUrl
-            upload_headers: Sequence[str],  # GraphQL type File.uploadHeaders
-            artifact_id: str,  # GraphQL type File.artifact.id
-        ) -> None:
-            pass
-
 
 # Request for a file to be prepared.
 class RequestPrepare(NamedTuple):
-    prepare_fn: "DoPrepareFn"
-    on_prepare: Optional["OnPrepareFn"]
+    file_spec: "internal_api.CreateArtifactFileSpecInput"
```

Comment:
Except for the `internal_api.Api` type signature, every other use of `internal_api` is for this long type signature and overall it would just feel more readable to me if we used `CreateArtifactFileSpecInput` directly. Up to you, but since you're adding all these in now it wouldn't change the diff size except by 1.

Reply:
Legit! Done, thanks for the suggestion!